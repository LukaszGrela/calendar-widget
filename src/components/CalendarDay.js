import React from 'react';
import PropTypes from 'prop-types';
import { momentObj } from 'react-moment-proptypes';

class CalendarDay extends React.Component {
    render = () => {
        const { date, className } = this.props;
        
        const text = typeof date === 'string' ? date : date.date();
        return (
            <div className={'day' + (className ? ' ' + className : '')}>{text}</div>
        )
    }
}
CalendarDay.propTypes = {
    date: PropTypes.oneOfType([PropTypes.string,momentObj]).isRequired
};
export default CalendarDay;