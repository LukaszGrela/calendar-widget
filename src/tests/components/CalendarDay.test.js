import React from 'react';
import moment from 'moment';
import { shallow } from 'enzyme';
import CalendarDay from '../../components/CalendarDay';

let wrapper, date, handleClick;

beforeEach(() => {
    date = moment();
    handleClick = jest.fn();
    wrapper = shallow(<CalendarDay
        date={date}
        handleClick={handleClick}
    />);

})
test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
});
test('should render correctly with text date value', () => {
    wrapper.setProps({
        date:'Thu'
    })
    expect(wrapper).toMatchSnapshot();
});

test('should handle click', () => {
    const div = wrapper.find('div.day');
    expect(div).toHaveLength(1);
    div.at(0).simulate('click');

    expect(handleClick).toHaveBeenLastCalledWith(moment())
});

test('should render correctly with given className', () => {
    wrapper.setProps({
        className:'my-class'
    })
    expect(wrapper).toMatchSnapshot();
});
