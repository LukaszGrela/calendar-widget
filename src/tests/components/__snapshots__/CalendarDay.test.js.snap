// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="day"
  onClick={[Function]}
>
  1
</div>
`;

exports[`should render correctly with given className 1`] = `
<div
  className="day my-class"
  onClick={[Function]}
>
  1
</div>
`;

exports[`should render correctly with text date value 1`] = `
<div
  className="day"
  onClick={[Function]}
>
  Thu
</div>
`;
